source/fundamentals/connection/tls.txt

```diff
@@ -152,6 +155,36 @@ please refer to the `JSSE Reference Guide
 
 .. _tls-disable-hostname-verification:
 
+Configure a Client-Specific Trust Store and Key Store
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can configure a client-specific trust store and key store using the
+``init()`` method of the ``SSLContext`` class.
+
+The following code snippet configures a client-specific trust store and key
+store:
+
+.. code-block:: java
+
+   String algorithm = "TLS";
+   try {
+      final SSLContext sslContext = SSLContext.getInstance(algorithm);
+      sslContext.init(<Your Key Managers>, <Your Trust Managers>, <A Secure Random Number Generator>);
+      MongoClientSettings settings = MongoClientSettings.builder()
+               .applyToSslSettings(builder ->
+                     builder.enabled(true)
+                              .context(sslContext)
+                              .build())
+               .build();
+      MongoClient client = MongoClients.create(settings);
+   }
+   catch (NoSuchAlgorithmException e) {
+      System.err.println("I'm sorry, but " + algorithm + " is not a valid message digest algorithm");
```

Comment:
Nit: I thought it was a bit awkward seeing code have personality.

```suggestion
      System.err.println(algorithm + " is not a valid message digest algorithm");
```

Reply:
I agree with this edit. This is awkward. Fix: https://github.com/mongodb/docs-java/pull/139/commits/b9860821c896c9999ff47eecf7ab8389175f8070

Was looking at examples of how people like to handle this error, and didn't look at the sample I was drawing from critically enough.